pkg/cluster/cluster.go

```diff
@@ -281,5 +287,110 @@ func New(ctx context.Context, log *logrus.Entry, _env env.Interface, db database
 		hiveClusterManager:                hiveClusterManager,
 		now:                               func() time.Time { return time.Now() },
 		openShiftClusterDocumentVersioner: new(openShiftClusterDocumentVersionerService),
-	}, nil
+	}
+
+	if doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		cloud := ""
+		switch strings.ToUpper(_env.Environment().Name) {
+		case dataplane.AzurePublicCloud:
+			cloud = dataplane.AzurePublicCloud
+		case dataplane.AzureUSGovCloud:
+			cloud = dataplane.AzureUSGovCloud
+		}
+
+		if cloud == "" {
+			return nil, errors.New("could not determine which Azure Cloud to use to instantiate MSI dataplane client")
+		}
```

Comment:
No, I don't think that will help. `_env.Environment()` already gives us the correct `AROEnvironment`.

The reason I had to put in this custom logic is because the cloud names are all uppercase in the [MSI dataplane module](https://github.com/Azure/msi-dataplane/blob/4e8e11fa5d231732524fbcfa65d96b98081eae7c/pkg/dataplane/constants.go#L7) but they're in camel case when you get them from the `AROEnvironment`.

I guess I could pull this out into a helper function so that you don't have to care about it as you're trying to read through and understand at a high-level how the cluster manager is being initialized.

Reply:
Sorry, yes you are right.
It sounds good to make a new helper function!